google/resource_compute_instance.go

```diff
@@ -378,22 +378,39 @@ func resourceComputeInstance() *schema.Resource {
 
 			"scheduling": &schema.Schema{
 				Type:     schema.TypeList,
+				MaxItems: 1,
 				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// The `scheduling` block is optional in the config but implicitly has a length
+					// of 1 with the default values for `automatic_restart` and `preemptible`.
+					// This suppress the diff `scheduling.#: "1" => "0"` when the `scheduling` block
+					// is not specified by the user.
+					if k == "scheduling.#" {
+						return true
+					}
+
+					// Don't suppress the diff for the scheduling fields.
+					return false
+				},
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"on_host_maintenance": &schema.Schema{
 							Type:     schema.TypeString,
 							Optional: true,
+							Computed: true,
 						},
 
 						"automatic_restart": &schema.Schema{
 							Type:     schema.TypeBool,
 							Optional: true,
```

Comment:
yeah, optional+computed is generally fine but booleans get weird since we can't tell the difference between `false` and unset. We need the user to be able to set `automatic_restart` to false, so we `ForceSendFields` it, which means that we'll always be providing a value to GCP and the computedness of it doesn't really end up mattering.